packages/block-editor/src/components/global-styles/typography-panel.js

```diff
@@ -252,6 +255,39 @@ export default function TypographyPanel( {
 		setFontAppearance( {} );
 	};
 
+	// Check if previous font style and weight values are available in the new font family
+	useEffect( () => {
```

Comment:
This effect runs when the component first mounts and will potentially create undo levels when `onChange` calls `setAttributes`. Is that what we want? 

I'm also running into a problem with some new code I'm writing where this effect is clobbering `style` because it's calling `onChange` on mount with an outdated `value`.

Is an effect necessary? Could the logic be in the callback that triggers when a new font family is selected?

Reply:
@noisysocks I think I've got something that avoids running the effect when the component first mounts - using a `useRef` to store the `fontFamily` value and then checking against that before running the logic. I've added this to the follow-up PR in this commit: https://github.com/WordPress/gutenberg/pull/63215/commits/18e1bfebf9b50cae3bb9fc81d1bf3814681b7922. Please let me know if this works better with what you're working on!

Reply:
Thanks I'll give that PR a spin!